File: app/src/main/java/com/android/mrt/mvp/java/contract/IMainContract.java
package com.android.mrt.mvp.java.contract;

import com.android.mrt.common.interfaces.ResultCallBack;

public interface IMainContract {

    interface IMainView {

        void showTitle(String title);

        void showUser(String user);
    }

    interface IMainPresenter {

        void getTitle();

        void getUser();
    }

    interface IMainModel {

        void getTitle(ResultCallBack<String> callback);
    }
}

File: app/src/main/java/com/android/mrt/mvp/contract/IMainContract.kt
package com.android.mrt.mvp.contract

import com.android.mrt.common.interfaces.ResultCallBack

interface IMainContract {

    interface IMainView {

        fun showTitle(title: String)

        fun showUser(user: String)
    }

    interface IMainPresenter {

        fun getTitle()

        fun getUser()
    }

    interface IMainModel {

        fun getTitle(callback: ResultCallBack<String>)
    }
}